### ISSUE TITLE
Using `--finetune_forecast` doesn't disable masking

### ISSUE THREAD

**Issue by kacpnowak** [ecmwf/WeatherGenerator#256]
### What happened?

 If in pretraining masking mode was used, using `--finetune_forecast` trains model for forecasting with masking still enabled. 

### What are the steps to reproduce the bug?

Train model in masking mode and then continue training using `--finetune_forecast` argument. Example output: 

![Image](https://github.com/user-attachments/assets/9dc14aaf-5c6a-4610-a5ec-89e079d886c8)

 

### Version

0.1

### Platform (OS and architecture)

Levante

### Relevant log output

```shell

```

### Organisation

AWI
